src/fcntl.rs

```diff
@@ -297,12 +304,12 @@ fn readlink_maybe_at<P: ?Sized + NixPath>(
                 cstr.as_ptr(),
                 v.as_mut_ptr() as *mut c_char,
                 v.capacity() as size_t,
-            ),
+            ) as _,
```

Comment:
`as _` can mask a wide range of type errors.  Could you please not do that?

Reply:
Unfortunately, [readlink](https://www.qnx.com/developers/docs/7.1/#com.qnx.doc.neutrino.lib_ref/topic/r/readlink.html) returns an `int` on QNX Neutrino. Any suggestion on how to resolve this (maybe one that is also applicable to `src/sys/select.rs`, see https://github.com/nix-rust/nix/pull/2055/files#r1265967722)?

One possibility (similar to the existing redox handling could be):
```rust
    path.with_nix_path(|cstr| unsafe {
        match dirfd {
            #[cfg(target_os = "redox")]
            Some(_) => unreachable!(),
            #[cfg(target_os = "nto")]
            Some(dirfd) => libc::readlinkat(
                dirfd,
                cstr.as_ptr(),
                v.as_mut_ptr() as *mut c_char,
                v.capacity() as size_t,
            ) as libc::ssize_t,
            #[cfg(not(any(target_os = "redox", target_os = "nto")))]
            Some(dirfd) => libc::readlinkat(
                dirfd,
                cstr.as_ptr(),
                v.as_mut_ptr() as *mut c_char,
                v.capacity() as size_t,
            ),
            #[cfg(target_os = "nto")]
            None => libc::readlink(
                cstr.as_ptr(),
                v.as_mut_ptr() as *mut c_char,
                v.capacity() as size_t,
            ) as libc::ssize_t,
            #[cfg(not(target_os = "nto"))]
            None => libc::readlink(
                cstr.as_ptr(),
                v.as_mut_ptr() as *mut c_char,
                v.capacity() as size_t,
            ),
        }
    })
```

But this makes the code a bit hard to read. Would a small OS specific wrapper function help?

Reply:
Either of those options would work, but I kind of like the first one better.

Reply:
Should be fixed, please check.